js/libs/core.js

```diff
@@ -617,21 +625,48 @@ Fliplet.FormBuilder = (function() {
       var hasOptions = component.props.options && Array.isArray(component.props.options.type());
       var hasSelectAll = component.props.addSelectAll && typeof component.props.addSelectAll.default === 'boolean';
       var isSlider = component.props._componentName.default === 'flSlider';
+      var isMatrixField = component.props._componentName.default === 'flMatrix';
 
       // If options is an array, automatically deal with options
-      if (hasOptions) {
-        component.computed._options = function generateOptions() {
-          return this.options.map(function(option) {
-            if (option.id && option.label && option.id != option.label) {
-              return option.label + ' <' + option.id + '>';
-            }
+      if (hasOptions || isMatrixField) {
+        if (isMatrixField) {
+          component.computed._rowOptions = function generateOptions() {
+            return this.rowOptions.map(function(option) {
+              if (option.id && option.label && option.id != option.label) {
+                return option.label + ' <' + option.id + '>';
+              }
+
+              return option.label || option.id;
+            }).join('\r\n');
+          };
 
-            return option.label || option.id;
-          }).join('\r\n');
-        };
+          component.computed._columnOptions = function generateOptions() {
+            return this.columnOptions.map(function(option) {
+              if (option.id && option.label && option.id != option.label) {
+                return option.label + ' <' + option.id + '>';
+              }
+
+              return option.label || option.id;
+            }).join('\r\n');
+          };
+        } else {
+          component.computed._options = function generateOptions() {
+            return this.options.map(function(option) {
+              if (option.id && option.label && option.id != option.label) {
+                return option.label + ' <' + option.id + '>';
+              }
+
+              return option.label || option.id;
+            }).join('\r\n');
+          };
+        }
+
+        component.methods._setOptions = function setOptions(str, attribute) {
+          if (!attribute) {
+            throw new Error('Attribute must be provided');
+          }
 
-        component.methods._setOptions = function setOptions(str) {
-          this.options = _.compact(str.split(/\r?\n/).map(function(rawOption) {
+          this[attribute] = _.compact(str.split(/\r?\n/).map(function(rawOption) {
```

Comment:
We should remove the lines with just empty space as well.

```suggestion
          this[attribute] = _.compact(_.map(str.split(/\r?\n/), function(option) {
            return option.trim();
          }).map(function(rawOption) {
```

Reply:
Done changes